MakieCore/src/basic_plots/lines.jl

```diff
@@ -0,0 +1,43 @@
+@plottype mutable struct Lines{N} <: AbstractPlot{Any}
+    position::AbstractVector{Point{N,Float32}}
```

Comment:
Yeah, that's up for debate... I would really like to keep it an AbstractVector, since there are quite a few use cases like efficiently mapping a view to a gpu buffer, or passing through GPUArrays and things like that...
But I guess we should benchmark the impact, and if it's huge, maybe try to stick to `Vector` as much as possible...

Reply:
Would it work to make the type of `positions` field a type-parameter of `Lines`?

Reply:
You can keep it `AbstractVector`. The trick is either (1) to not pull it out of the `Lines` struct until the "last minute" and/or (2) annotate dispatches with it with `@nospecialize` until you deliberately want a function barrier (basically, for any method with a loop over the elements).

Reply:
> Would it help to make the type of positions field a type-parameter of Lines?

It would improve inferrability at the cost of likely greater latency, because the compiler would have to specialize every call for every `Scatter` type you use.